readthedocs/oauth/services/gitlab.py

```diff
@@ -0,0 +1,235 @@
+"""OAuth utility functions"""
+
+import logging
+import json
+import re
+
+from django.conf import settings
+from requests.exceptions import RequestException
+from allauth.socialaccount.models import SocialToken
+from allauth.socialaccount.providers.gitlab.views import GitLabOAuth2Adapter
+from urlparse import urljoin
+
+from readthedocs.restapi.client import api
+
+from ..models import RemoteOrganization, RemoteRepository
+from .base import Service, DEFAULT_PRIVACY_LEVEL
+
+
+log = logging.getLogger(__name__)
+
+
+class GitLabService(Service):
+    """Provider service for GitLab"""
+
+    adapter = GitLabOAuth2Adapter
+    url_pattern = re.compile(re.escape(adapter.provider_base_url))
+    default_avatar = {
+        'repo': urljoin(settings.MEDIA_URL, 'images/fa-bookmark.svg'),
+        'org':  urljoin(settings.MEDIA_URL, 'images/fa-users.svg'),
+    }
+
+    def paginate(self, url, **kwargs):
+        """Combines return from GitLab pagination. GitLab uses
+        LinkHeaders, see: http://www.w3.org/wiki/LinkHeader
+
+        :param url: start url to get the data from.
+        :param kwargs: optional parameters passed to .get() method
+
+        See http://doc.gitlab.com/ce/api/README.html#pagination
```

Comment:
This URL should be updated to `https://docs.gitlab.com/ce/api/README.html#pagination`